task/host-sp-comms/src/bsp/cosmo_a.rs

```diff
@@ -102,17 +102,16 @@ impl ServerImpl {
             //
             // Sharkfin connectors start at J200 and are numbered sequentially
             4..=13 => {
-                let (designator, f): ([u8; 4], _) =
+                let (designator, f): ([u8; 7], _) =
                     Self::get_sharkfin_vpd(index as usize - 4);
                 let mut name = *b"____/U2/ID";
-                name[0..4].copy_from_slice(&designator);
+                name[0..4].copy_from_slice(&designator[0..4]);
                 self.read_eeprom_barcode(sequence, &name, f)
             }
             14..=23 => {
-                let (designator, f): ([u8; 4], _) =
+                let (mut name, f): ([u8; 7], _) =
                     Self::get_sharkfin_vpd(index as usize - 14);
-                let mut name = *b"____/U2";
-                name[0..4].copy_from_slice(&designator);
+                name[4] = b'/';
```

Comment:
If you change `const fn byteify` to translate from `_` to `/`, you could make this a lot more foolproof!

Reply:
good call! i didn't mess with this code too much since i was hoping to change it to just use the IDs from codegen in a subsequent PR, but I think this is a good middle ground.